portal-web/test/functional/com/liferay/portalweb/macros/json/remote_apps/JSONRemoteAppAPI.macro

```diff
@@ -31,25 +31,18 @@ definition {
 		var portalURL = JSONCompany.getPortalURL();
 
 		var curl = '''
-			${portalURL}/api/jsonws/remoteapp.clientextensionentry/add-custom-element-client-extension-entry \
+			${portalURL}/api/jsonws/remoteapp.clientextensionentry/add-client-extension-entry \
 				  -u test@liferay.com:test \
-				  -d customElementCSSURLs=${customElementCssurlModified} \
-				  -d customElementHTMLElementName=${htmlNameModified} \
-				  -d customElementURLs=${customElementURLModified} \
-				  -d customElementUseESM=${customElementUseESMValue} \
 				  -d description= \
 				  -d externalReferenceCode= \
-				  -d friendlyURLMapping= \
-				  -d instanceable=true \
 				  -d nameMap='{\"en_US\":\"${nameModified}\"}' \
-				  -d portletCategoryName=RemoteApps \
 				  -d properties=${customElementPropertiesModified} \
-				  -d sourceCodeURL=
+				  -d sourceCodeURL= \
+				  -d type='customElement' \
+				  -d typeSettings=cssURLs=${customElementCssurlModified}%0AhtmlElementName=${htmlNameModified}%0Aurls=${customElementURLModified}%0AuseESM=${customElementUseESMValue}%0Ainstanceable=true%0AportletCategoryName=category.remote-apps
```

Comment:
I feel like I should break this up.. it's a little long.... :eyes:

Reply:
Perhaps can hide the list like in JSONAssetlistSetter.setTypeSettings. Instead of selectionType dynamic set it to `customElement`

```
			if ("${selectionType}" == "dynamic") {
				var typeSettings = '''
anyAssetType=true
classNameIds=${classNameIds}
classTypeIdsDLFileEntryAssetRendererFactory=${classTypeIdsDLFileEntry}
classTypeIdsJournalArticleAssetRendererFactory=${classTypeIdsDDMStructure}
groupIds=${groupId}
orderByColumn1=modifiedDate
orderByColumn2=title
orderByType1=DESC
orderByType2=ASC
queryAndOperator0=true
queryContains0=true
queryName0=assetTags
subtypeFieldsFilterEnabledDLFileEntryAssetRendererFactory=false
subtypeFieldsFilterEnabledJournalArticleAssetRendererFactory=false
''';


		var newLine = '''
''';

		var typeSettings = StringUtil.regexReplaceAll("${typeSettings}", "${newLine}", "%0A");

		return "${typeSettings}";
```

Just something I saw, I have no idea what is a good standard is curl and JSON macro poshi world.